ios/RIBs/Classes/DI/Component.swift

```diff
@@ -47,11 +47,7 @@ open class Component<DependencyType>: Dependency {
             lock.unlock()
         }
 
-        // Additional nil coalescing is needed to mitigate a Swift bug appearing in Xcode 10.
-        // see https://bugs.swift.org/browse/SR-8704.
-        // Without this measure, calling `shared` from a function that returns an optional type
-        // will always pass the check below and return nil if the instance is not initialized.
-        if let instance = (sharedInstances[__function] as? T?) ?? nil {
```

Comment:
Removing this would break applications on Swift 4.2. Can we guard this with `#if swift(>=5.0)`

Reply:
@sbarow Done :)